ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -802,48 +802,67 @@ func (r *RayServiceReconciler) reconcileIngress(ctx context.Context, rayServiceI
 }
 
 func (r *RayServiceReconciler) reconcileServices(ctx context.Context, rayServiceInstance *rayv1alpha1.RayService, rayClusterInstance *rayv1alpha1.RayCluster, serviceType common.ServiceType) error {
-	// Creat Service Struct.
-	var raySvc *corev1.Service
+	r.Log.Info(
+		"reconcileServices", "serviceType", serviceType,
+		"RayService name", rayServiceInstance.Name, "RayService namespace", rayServiceInstance.Namespace,
+	)
+
+	var newSvc *corev1.Service
 	var err error
-	if serviceType == common.HeadService {
-		raySvc, err = common.BuildHeadServiceForRayService(*rayServiceInstance, *rayClusterInstance)
-	} else if serviceType == common.ServingService {
-		raySvc, err = common.BuildServeServiceForRayService(*rayServiceInstance, *rayClusterInstance)
+
+	switch serviceType {
+	case common.HeadService:
+		newSvc, err = common.BuildHeadServiceForRayService(*rayServiceInstance, *rayClusterInstance)
+	case common.ServingService:
+		newSvc, err = common.BuildServeServiceForRayService(*rayServiceInstance, *rayClusterInstance)
+	default:
+		return fmt.Errorf("unknown service type %v", serviceType)
 	}
 
+	// TODO: This seems to be useless because BuildHeadServiceForRayService and BuildServeServiceForRayService will not return error.
 	if err != nil {
 		return err
 	}
-	raySvc.Name = utils.CheckName(raySvc.Name)
+	r.Log.Info("reconcileServices", "newSvc", newSvc)
 
-	// Get Service instance.
-	rayService := &corev1.Service{}
-	err = r.Get(ctx, client.ObjectKey{Name: raySvc.Name, Namespace: rayServiceInstance.Namespace}, rayService)
+	// Retrieve the Service from the Kubernetes cluster with the name and namespace.
+	oldSvc := &corev1.Service{}
+	err = r.Get(ctx, client.ObjectKey{Name: newSvc.Name, Namespace: rayServiceInstance.Namespace}, oldSvc)
 
 	if err == nil {
-		// Update Service
-		rayService.Spec = raySvc.Spec
-		r.Log.V(1).Info("reconcileServices update service")
-		if updateErr := r.Update(ctx, rayService); updateErr != nil {
-			r.Log.Error(updateErr, "raySvc Update error!", "raySvc.Error", updateErr)
+		// Only update the service if the RayCluster switches.
```

Comment:
Is "Only update the service if the RayCluster switches" a new behavior change in this PR? What's the reasoning behind it? If the reason is easy to state it might be worth adding it to this code comment.

Reply:
> Is "Only update the service if the RayCluster switches" a new behavior change in this PR?

Yes

> What's the reasoning behind it? If the reason is easy to state it might be worth adding it to this code comment.

Similar to #1065, the redundant `Update` invocations will cause an unnecessary burden on the Kubernetes API Server.